### PR TITLE
A REST layer built with akka-http and spray-json.

### PR DESCRIPTION
**Pull request by dragos**
It’s a separate project with its own Sbt build in order to use
Scala 2.11 and the fancy new akka-http module.

### CODE REVIEW

**Comment by dragos** [comment]
@dotta please have a look

File: rest-api/src/main/scala/pipeline/rest/Main.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        complete {
+          PipelineCluster.allRecords()
+        }
+      }
+
+    }
+
+  Http().bindAndHandle(route, "localhost", 1111)
+  println("Press ENTER to terminate")
+  Console.readLine()
```
Comment (dotta) [inline rest-api/src/main/scala/pipeline/rest/Main.scala]: If this throws an exception the actor system won't be shut down

File: rest-api/src/main/scala/pipeline/rest/Main.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import akka.http.scaladsl.marshallers.xml.ScalaXmlSupport._
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.model._
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import akka.util.ByteString
+import pipeline.data.PipelineCluster
+import scala.collection.JavaConverters._
+import spray.json._
+import pipeline.data._
```
Comment (dotta) [inline rest-api/src/main/scala/pipeline/rest/Main.scala]: do you really need all these imports (e.g., Await and Duration don't seem to be needed)

File: rest-api/src/main/scala/pipeline/data/PipelineCluster.scala [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import com.datastax.driver.core.Cluster
+import com.datastax.driver.core.PreparedStatement
+import com.datastax.driver.core.ResultSet
+import com.typesafe.config.ConfigFactory
+
+import scala.collection.JavaConverters._
+
+object PipelineCluster {
+  val config = ConfigFactory.load()
```
Comment (dotta) [inline rest-api/src/main/scala/pipeline/data/PipelineCluster.scala:14]: The `config` could be passed in argument to your main (and, if none is provided, you could default to `ConfigFactory.load`)

**Comment by dotta** [comment]
Full disclosure: @dragos explicitly asked for nitpicking, so I'm doing my best :-)

File: rest-api/build.sbt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name := "rest-api"
+
+scalaVersion := "2.11.7"
+
+resolvers += 
+	"Websudos releases" at "https://dl.bintray.com/websudos/oss-releases/"
+
+val phantomVersion = "1.12.2"
```
Comment (dotta) [inline rest-api/build.sbt]: Where is this used?

File: rest-api/src/main/scala/pipeline/data/PipelineCluster.scala [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  /** A prepared statement for range queries. Only bound values are shipped for each query. */
+  lazy val rangeQuery: PreparedStatement =
+    session.prepare(s"""|SELECT * from $alleleTable
+                        |WHERE population = ?
+                        |AND chromosome = ?
+                        |AND start > ?
+                        |AND start < ? ;""".stripMargin)
+
+  def rangeQuery(pop: String, chromosome: String, start: Int, end: Int): Seq[AlleleRecord] = {
```
Comment (dotta) [inline rest-api/src/main/scala/pipeline/data/PipelineCluster.scala:44]: Would it be preferable to return an immutable Seq?

**Comment by dotta** [comment]
Looks decent to me :-P

**Comment by dragos** [comment]
ok, let's merge this! thanks @dotta !

**Comment by dotta** [comment]
Any time! Good job man.

**Pull request merged**